kolibri/core/logger/tasks.py

```diff
@@ -19,48 +20,39 @@ def get_logs_dir_and_filepath(log_type, facility):
     return logs_dir, filepath
 
 
-def validate_startexportlogcsv(request, request_data):
-    facility_id = request_data.get("facility", None)
-    if facility_id:
-        facility = Facility.objects.get(pk=facility_id)
-    else:
-        facility = request.user.facility
-
-    log_type = request_data.get("logtype", "summary")
-
-    if log_type in CSV_EXPORT_FILENAMES.keys():
-        logs_dir, filepath = get_logs_dir_and_filepath(log_type, facility)
-    else:
-        raise Http404("Impossible to create a csv export file for {}".format(log_type))
-
-    if not os.path.isdir(logs_dir):
-        os.mkdir(logs_dir)
-
-    job_type = "EXPORTSUMMARYLOGCSV" if log_type == "summary" else "EXPORTSESSIONLOGCSV"
+class ExportLogCSVValidator(JobValidator):
+    facility = serializers.PrimaryKeyRelatedField(
+        queryset=Facility.objects.all(), required=False
+    )
+    log_type = serializers.ChoiceField(choices=list(CSV_EXPORT_FILENAMES.keys()))
 
-    job_metadata = {
-        "type": job_type,
-        "started_by": request.user.pk,
-        "facility": facility.id,
-    }
+    def validate(self, data):
+        facility = data.get("facility", None)
+        if facility is None and "user" in self.context:
+            facility = self.context["user"].facility
+        elif facility is None:
+            raise serializers.ValidationError(
+                "Facility must be specified when no user is available."
+            )
+        logs_dir, filepath = get_logs_dir_and_filepath(data["log_type"], facility)
+        if not os.path.isdir(logs_dir):
+            os.mkdir(logs_dir)
 
-    return {
-        "log_type": log_type,
-        "filepath": filepath,
-        "facility": facility,
-        "extra_metadata": job_metadata,
-    }
+        return {
```

Comment:
I don't see here some of the data that was in the original implementation. 
The previous code provided to  the frontend  "`started_by`",  "`type`"  and the whole "`facility`" object. Are these values now returned by default by the `JobValidator` always?

Reply:
I noticed that `started_by` and `started_by_username` were being added in a lot of tasks, so I moved them into the base `JobValidator` class, so that it always gets added to every task in the extra_metadata.

"type" is now just the import path of the task (so this will require updates to frontend constants) so for this it would be `kolibri.core.logger.tasks.exportuserscsv` - this has the nice side effect that the type of the task is now used to start the task in question and then refer to it later when in progress.

I was confused as to how the `facility` object was being passed around, as it is not JSON serializable. There is now a nullable `facility_id` property on every job object, so it seemed reasonable to use that to refer to which facility the job is being run on.